src/main/scala/viper/gobra/translator/encodings/IntEncoding.scala

```diff
@@ -48,6 +64,103 @@ class IntEncoding extends LeafTypeEncoding {
       case e@ in.Mul(l, r) => for {vl <- goE(l); vr <- goE(r)} yield withSrc(vpr.Mul(vl, vr), e)
       case e@ in.Mod(l, r) => for {vl <- goE(l); vr <- goE(r)} yield withSrc(vpr.Mod(vl, vr), e)
       case e@ in.Div(l, r) => for {vl <- goE(l); vr <- goE(r)} yield withSrc(vpr.Div(vl, vr), e)
+
+      case e@ in.BitAnd(l, r) => for {vl <- goE(l); vr <- goE(r)} yield withSrc(vpr.FuncApp(bitwiseAnd, Seq(vl, vr)), e)
+      case e@ in.BitOr(l, r)  => for {vl <- goE(l); vr <- goE(r)} yield withSrc(vpr.FuncApp(bitwiseOr,  Seq(vl, vr)), e)
+      case e@ in.BitXor(l, r) => for {vl <- goE(l); vr <- goE(r)} yield withSrc(vpr.FuncApp(bitwiseXor, Seq(vl, vr)), e)
+      case e@ in.BitClear(l, r)   => for {vl <- goE(l); vr <- goE(r)} yield withSrc(vpr.FuncApp(bitClear, Seq(vl, vr)), e)
+      case e@ in.ShiftLeft(l, r)  => withSrc(handleShift(shiftLeft)(l, r), e)
+      case e@ in.ShiftRight(l, r) => withSrc(handleShift(shiftRight)(l, r), e)
+      case e@ in.BitNeg(exp)  => for {ve <- goE(exp)} yield withSrc(vpr.FuncApp(bitwiseNegation, Seq(ve)), e)
     }
   }
-}
+
+  override def finalize(col: Collector): Unit = {
+    col.addMember(bitwiseAnd)
+    col.addMember(bitwiseOr)
+    col.addMember(bitwiseXor)
+    col.addMember(bitClear)
+    col.addMember(shiftLeft)
+    col.addMember(shiftRight)
+    col.addMember(bitwiseNegation)
+  }
+
+  /* Bitwise Operations */
+  private val bitwiseAnd: vpr.Function =
```

Comment:
A few comments to this file:
- all bitwise operations are simply uninterpreted function applications, right? Thus, connecting to bit vectors comes at a later point in time?
- you could change your bitwise operations to be `lazy val` and keep track whether they are actually invoked such that only used functions get emitted

Reply:
> What is the cardinality operator for multisets now?

It is the same as the regular sets: `len(exp)` where `exp`

Reply:
That is indeed a good idea